File: src/m/qmu/param_write.py
#move this later:
from helpers import *


def param_write(fidi, sbeg, pname, smid, s, params):
    '''
function to write a parameter
'''
    if not isfield(params, pname):
        print('WARNING: param_write:param_not_found: Parameter {} not found in structure.'.format(pname))
        return

    params_pname = vars(params)[pname]

    if type(params_pname) == bool and (not params_pname):
        return

    if type(params_pname) == bool:
        fidi.write(sbeg + str(pname) + s)

    elif type(params_pname) in [str]:
        fidi.write(sbeg + pname + smid + params_pname + s)

    elif type(params_pname) in [int, float]:
        fidi.write(sbeg + str(pname) + smid + str(params_pname) + s)

File: src/m/qmu/param_write.m
%
%  function to write a parameter
%
function []=param_write(fidi,sbeg,pname,smid,send,params)

if ~isfield(params,pname)
    warning('param_write:param_not_found',...
        'Parameter ''%s'' not found in structure.',pname);
    return
end

if islogical(params.(pname)) && ~params.(pname)
    return
end

if     islogical(params.(pname))
    fprintf(fidi,[sbeg '%s' send],pname);
elseif ischar   (params.(pname))
    fprintf(fidi,[sbeg '%s' smid '%s' send],pname,params.(pname));
elseif isnumeric(params.(pname))
    fprintf(fidi,[sbeg '%s' smid '%g' send],pname,params.(pname));
end

end
